sdk/core/Azure.Core/src/Pipeline/HttpWebRequestTransport.cs

```diff
@@ -0,0 +1,302 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+namespace Azure.Core.Pipeline
+{
+#if NETFRAMEWORK
+    /// <summary>
+    /// The <see cref="HttpWebRequest"/> based <see cref="HttpPipelineTransport"/> implementation.
+    /// </summary>
+    internal class HttpWebRequestTransport : HttpPipelineTransport
+    {
+        public static readonly HttpWebRequestTransport Shared = new HttpWebRequestTransport();
+        private readonly IWebProxy? _proxy;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="HttpWebRequestTransport"/>
+        /// </summary>
+        public HttpWebRequestTransport()
+        {
+            if (HttpEnvironmentProxy.TryCreate(out IWebProxy webProxy))
+            {
+                _proxy = webProxy;
+            }
+        }
+
+        /// <inheritdoc />
+        public override void Process(HttpMessage message)
+        {
+            ProcessInternal(message, false).EnsureCompleted();
+        }
+
+        /// <inheritdoc />
+        public override async ValueTask ProcessAsync(HttpMessage message)
+        {
+            await ProcessInternal(message, true);
+        }
+
+        private async ValueTask ProcessInternal(HttpMessage message, bool async)
+        {
+            var request = CreateRequest(message.Request);
+            using var registration = message.CancellationToken.Register(state => ((HttpWebRequest) state).Abort(), request);
+            try
+            {
+                if (message.Request.Content != null)
+                {
+                    if (request.ContentLength == -1 &&
```

Comment:
Would it be safer to just check `< 0`?

Reply:
I'll keep it -1 to know if there are any other magic numbers :)